command/alias.go

```diff
@@ -0,0 +1,98 @@
+package command
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cli/cli/utils"
+	"github.com/google/shlex"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	RootCmd.AddCommand(aliasCmd)
+	aliasCmd.AddCommand(aliasSetCmd)
+}
+
+var aliasCmd = &cobra.Command{
+	Use:   "alias",
+	Short: "Create shortcuts for gh commands",
+	Long:  `TODO`,
+}
+
+var aliasSetCmd = &cobra.Command{
+	Use: "set <alias> <expansion>",
+	// TODO HACK: even when inside of a single-quoted string, cobra was noticing and parsing any flags
+	// used in an alias expansion string argument. Since this command needs no flags, I disabled their
+	// parsing. If we ever want to add flags to alias set we'll have to figure this out. I haven't
+	// checked if this is fixed in a new cobra.
```

Comment:
`gh alias set pc 'pr create --assignee mona'` is exactly what _didn't_ work, hence my saying "even when inside of a single-quoted string" and saying it seemed like a bug.

I understand how argument passing works :)

Reply:
> `gh alias set pc 'pr create --assignee mona'` is exactly what _didn't_ work

I only brought up my points because I have tried to reproduce this with `DisableFlagParsing: false` and failed.

Reply:
this is my experience with `DisableFlagParsing` unset:

![image](https://user-images.githubusercontent.com/98482/83044527-e8da6c80-a009-11ea-9654-912b3fc4bf0c.png)

Reply:
I had trouble reproducing this too. It might be that you’re trying to use the `--author` flag on pr list but it surprisingly doesn’t exist. I tried out these alias for issues and they worked:

`ghd alias set foo ‘issue list —author=“probablycorey”’`
`ghd alias set foo2 ‘issue list —author=probablycorey’`

Reply:
the error is from `alias set`; not `pr list`. i get the same result with issue list:

![image](https://user-images.githubusercontent.com/98482/83066809-3d421400-a02b-11ea-98e9-a0446136b2ea.png)

I thought it was possibly something weird about my shell but I'm getting the same behavior.

I noticed that y'all aren't using actual single quotes but rather the  ‘ character. I tried that but it still fails for me; I wonder, however, if that character is treated different on macos?

Reply:
@vilmibm I'm stumped. I cannot reproduce this when I remove the `DisableFlagParsing` workaround:
```sh
$ gh alias set ix 'issue list --author=vilmibm'
- Adding alias for ix: issue list --author=vilmibm
✓ Added alias.

$ gh alias set ix 'issue list --author="vilmibm"'
- Adding alias for ix: issue list --author="vilmibm"
✓ Changed alias ix from issue list --author=vilmibm to issue list --author="vilmibm"

$ gh ix

Showing 11 of 11 issues in cli/cli that match your search

#990  gh alias delete                     (aliases, enhancement)  about 6 days ago
...
```

Reply:
:shrug: i have no further ideas. it is concretely broken for me. i think we should leave flag parsing disabled until we have a reason to worry about it; i'll amend my comment to mention the weird by-platform behavior.

Reply:
@vilmibm I can see that you are running `ghd` instead of `gh`. Could you check in your script that you are forwading the arguments with `"$@"` and not via `$@` or `$*`?